upb/def.c

```diff
@@ -721,6 +722,35 @@ const char *upb_fielddef_name(const upb_fielddef *f) {
   return upb_def_fullname(upb_fielddef_upcast(f));
 }
 
+bool upb_fielddef_getjsonname(const upb_fielddef *f, char *buf) {
```

Comment:
Camel casing the regular name is the *default* for the JSON name, but it can also explicitly be set to something else. See: https://github.com/google/protobuf/blob/master/src/google/protobuf/descriptor.proto#L193

Separating out as a utility function could possibly be useful, but I'd rather late-bind on that.